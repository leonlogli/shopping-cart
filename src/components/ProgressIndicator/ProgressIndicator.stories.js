import React from 'react'

import ProgressIndicator from './ProgressIndicator'

export default {
  title: 'ProgressIndicator',
}

export const Default = () => <ProgressIndicator />

export const circular = () => <ProgressIndicator circular />

export const centered = () => <ProgressIndicator position="center" />
