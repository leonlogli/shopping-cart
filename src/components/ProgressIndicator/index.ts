export * from './ProgressIndicator'
export { default } from './ProgressIndicator'
