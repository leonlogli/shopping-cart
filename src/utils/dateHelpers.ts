const durationFormat = (interval: number, type: string) => {
  return `il y a ${Math.floor(interval)} ${type}${
    Math.floor(interval) > 1 ? 's' : ''
  }`
}

const fromNow = (date: Date) => {
  const seconds = Math.floor(
    ((new Date() as any) - (new Date(date) as any)) / 1000
  )

  let interval = seconds / 31536000

  if (Math.floor(interval) >= 1) {
    return durationFormat(interval, 'an')
  }

  interval = seconds / 2592000
  if (Math.floor(interval) >= 1) {
    return `il y a ${Math.floor(interval)} mois`
  }

  interval = seconds / 86400
  if (Math.floor(interval) >= 1) {
    return durationFormat(interval, 'jour')
  }

  interval = seconds / 3600
  if (Math.floor(interval) >= 1) {
    return durationFormat(interval, 'heure')
  }

  interval = seconds / 60
  if (Math.floor(interval) >= 1) {
    return durationFormat(interval, 'minute')
  }

  return `Ã  l'instant`
}

/**
 * Parse the given string time to a date.Time must have `hh:mm:ss` format
 * @param time time to parse
 * @param date date to which the time will be set once parsed. If null, today's date is used
 */
const parseTimeToDate = (time: string, date?: Date) => {
  const dateToSet = date || new Date()
  const timeParts = time.split(':').map((t) => Number(t) || 0)

  dateToSet.setHours(timeParts[0], timeParts[1], timeParts[2] || 0)

  return dateToSet
}

const formatShortDate = (date: any, locale?: string) => {
  return new Date(date).toLocaleDateString(locale, {
    day: 'numeric',
    month: 'short',
    year: 'numeric',
  })
}

export { fromNow, parseTimeToDate, formatShortDate }
