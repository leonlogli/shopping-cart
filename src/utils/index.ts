export * from './testHelpers'
export * from './util'
export * from './dateHelpers'
