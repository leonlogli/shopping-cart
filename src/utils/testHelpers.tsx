import React from 'react'
import { ThemeProvider } from 'styled-components'
import { mount, shallow } from 'enzyme'
import { defaultTheme } from '../styles'

const ThemeProviderWrapper = ({ children }: any) => (
  <ThemeProvider theme={defaultTheme}>{children}</ThemeProvider>
)

export const shallowWithTheme = (tree: any) =>
  shallow(tree, {
    wrappingComponent: ThemeProviderWrapper,
  })

export const mountWithTheme = (tree: any) =>
  mount(tree, {
    wrappingComponent: ThemeProviderWrapper,
  })
