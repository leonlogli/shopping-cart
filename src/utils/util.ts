const toUSD = (value: number) => {
  return `${value.toLocaleString('en-US', {
    minimumFractionDigits: 2,
  })} USD`
}

export { toUSD }
