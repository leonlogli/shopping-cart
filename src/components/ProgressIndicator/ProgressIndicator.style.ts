import styled, { css } from 'styled-components'
import MuiCircularProgress from '@material-ui/core/CircularProgress'
import MuiLinearProgress from '@material-ui/core/LinearProgress'

import { ProgressIndicatorProps } from './ProgressIndicator'

const positionStyle = (position: ProgressIndicatorProps['position']) => {
  if (position === 'top') {
    return css`
      position: fixed;
      z-index: 1032;
      top: 0;
      left: 0;
    `
  }

  return css`
    position: fixed;
    z-index: 1032;
    top: 50vh;
    left: 50vw;
  `
}

const CircularProgress = styled(MuiCircularProgress)<ProgressIndicatorProps>`
  ${({ position }) => positionStyle(position)};
`

const LinearProgress = styled(MuiLinearProgress)<ProgressIndicatorProps>`
  height: 3px;
  width: 100%;
  ${({ position }) => positionStyle(position)};
  left: 0;
`

export { CircularProgress, LinearProgress }
