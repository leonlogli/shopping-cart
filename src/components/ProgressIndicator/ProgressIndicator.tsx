import React from 'react'
import { CircularProgressProps } from '@material-ui/core/CircularProgress'
import { LinearProgressProps } from '@material-ui/core/LinearProgress'

import { CircularProgress, LinearProgress } from './ProgressIndicator.style'

export type ProgressIndicatorProps = CircularProgressProps &
  LinearProgressProps & {
    /**
     * Whether the indicator should be circular
     * @default false
     */
    circular?: boolean
    /**
     * The progress indicator position
     * @default 'top' for linear and 'center' for circular indicator
     */
    position?: 'top' | 'center'
  }

/**
 * Indicates the progress of an operation
 */
const ProgressIndicator = (props: ProgressIndicatorProps) => {
  const { circular, position = circular ? 'center' : 'top', ...other } = props

  if (circular) {
    return <CircularProgress position={position} {...other} />
  }

  return <LinearProgress position={position} {...other} />
}

export { ProgressIndicator }
export default ProgressIndicator
