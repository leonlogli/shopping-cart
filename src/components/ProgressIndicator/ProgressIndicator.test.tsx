import React from 'react'
import 'jest-styled-components'

import { mountWithTheme } from '../../utils'

import ProgressIndicator from './ProgressIndicator'

describe('<ProgressIndicator />', () => {
  const wrapper = mountWithTheme(<ProgressIndicator />)

  it('should render children', () => {
    expect(wrapper.find('.MuiLinearProgress-root').length).toEqual(1)
  })

  it('should match a snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  })

  it('should set be circular', () => {
    wrapper.setProps({ circular: true } as any)

    expect(wrapper.find('.MuiCircularProgress-root').length).toEqual(1)
  })

  it('should be positioned at the top', () => {
    wrapper.setProps({ position: 'top' } as any)

    expect(wrapper).toHaveStyleRule('top', '0')
  })

  it('should be centered', () => {
    wrapper.setProps({ position: 'center' } as any)

    expect(wrapper).toHaveStyleRule('top', '50vh')
    expect(wrapper).toHaveStyleRule('left', '50vw')
  })
})
