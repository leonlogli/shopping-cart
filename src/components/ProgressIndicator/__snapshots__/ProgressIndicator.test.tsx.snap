// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ProgressIndicator /> should match a snapshot 1`] = `
<ProgressIndicator>
  <Styled(WithStyles(ForwardRef(LinearProgress)))
    position="top"
  >
    <WithStyles(ForwardRef(LinearProgress))
      className="jicwCj"
      position="top"
    >
      <ForwardRef(LinearProgress)
        className="jicwCj"
        classes={
          Object {
            "bar": "MuiLinearProgress-bar",
            "bar1Buffer": "MuiLinearProgress-bar1Buffer",
            "bar1Determinate": "MuiLinearProgress-bar1Determinate",
            "bar1Indeterminate": "MuiLinearProgress-bar1Indeterminate",
            "bar2Buffer": "MuiLinearProgress-bar2Buffer",
            "bar2Indeterminate": "MuiLinearProgress-bar2Indeterminate",
            "barColorPrimary": "MuiLinearProgress-barColorPrimary",
            "barColorSecondary": "MuiLinearProgress-barColorSecondary",
            "buffer": "MuiLinearProgress-buffer",
            "colorPrimary": "MuiLinearProgress-colorPrimary",
            "colorSecondary": "MuiLinearProgress-colorSecondary",
            "dashed": "MuiLinearProgress-dashed",
            "dashedColorPrimary": "MuiLinearProgress-dashedColorPrimary",
            "dashedColorSecondary": "MuiLinearProgress-dashedColorSecondary",
            "determinate": "MuiLinearProgress-determinate",
            "indeterminate": "MuiLinearProgress-indeterminate",
            "query": "MuiLinearProgress-query",
            "root": "MuiLinearProgress-root",
          }
        }
        position="top"
      >
        <div
          className="MuiLinearProgress-root MuiLinearProgress-colorPrimary jicwCj MuiLinearProgress-indeterminate"
          position="top"
          role="progressbar"
        >
          <div
            className="MuiLinearProgress-bar MuiLinearProgress-barColorPrimary MuiLinearProgress-bar1Indeterminate"
            style={Object {}}
          />
          <div
            className="MuiLinearProgress-bar MuiLinearProgress-bar2Indeterminate MuiLinearProgress-barColorPrimary"
            style={Object {}}
          />
        </div>
      </ForwardRef(LinearProgress)>
    </WithStyles(ForwardRef(LinearProgress))>
  </Styled(WithStyles(ForwardRef(LinearProgress)))>
</ProgressIndicator>
`;
